book-store-inventory/pkg/repositories/book_repository.go

```diff
@@ -50,7 +51,12 @@ func (bookModel *BookModel) GetBooks() ([]models.Book, error) {
 }
 
 func (bookModel *BookModel) GetBookByID(bookID string) (models.Book, error) {
-	statement := `SELECT book_id, title, author, price, published_date, stock_quantity FROM books WHERE book_id = ? ;`
+	// Input validation
+	if bookID == "" {
+		return models.Book{}, fmt.Errorf("book ID cannot be empty")
```

Comment:
we should handle custom errors if any param is empty, maybe return a model which say things are empty so that it can be handled on the handler side